aten/src/ATen/native/TensorShape.cpp

```diff
@@ -909,9 +909,27 @@ Tensor view(const Tensor& self, IntArrayRef size) {
     "not compatible with input tensor's size and stride (at least one dimension"
     " spans across two contiguous subspaces). Use .reshape(...) instead.");
   auto stride_value = *stride;
-  auto self_ = self.clone();
-  self_.set_(self.storage(), self.storage_offset(), inferred_size,
-             stride_value);
+  auto self_ = self.alias();
```

Comment:
I can't comment on the above, but this also looks wrong:
```
  if (self.sizes() == inferred_size) {
    return self;
}
```

I don't know if you can induce an error from the python frontend because we have a bunch of extra code for Variable-scope views, but in non-Variable mode, doesn't this mean that:
```
Tensor x = at::randn(2,3);
y = x.view(2,3);
x.resize_(3,2);
// y will be resized even though it's a view!
```

can you verify this / add a test?

Reply:
yeah this looks wrong.

Reply:
just added some tests, looks like it `y` didn't got resized.
maybe it's because we pass in const Tensor& and when we return `self` it got copied? but it should just copy the `intrusive_ptr` rather than `TensorImpl`?

I saw Tensor has default copy constructor:
`Tensor(const Tensor&) = default;`, does it copy the underlying `impl_`: `c10::intrusive_ptr<TensorImpl, UndefinedTensorImpl> impl_;`?

Reply:
`Tensor` doesn't matter, what matters if there is a different `TensorImpl` or not.  I think you'd have to write an `at::` test to find that bug, we have tests like that in `/aten/src/ATen/test/`.

Reply:
yeah testing tensor directly will catch the bug.